From 2a0bb5b2663f1bdf6b9c7ad0e3134003c65903ab Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael.luceno@gmail.com>
Date: Sun, 5 May 2013 20:14:24 -0300
Subject: [PATCH] Fix NULL pointer dereference

When cpufreq is not available, cpufreq_get_related_cpus returns NULL.

Signed-off-by: Ismael Luceno <ismael.luceno@gmail.com>
---
 utils/set.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/utils/set.c b/utils/set.c
index bfca465..5b51da1 100644
--- a/utils/set.c
+++ b/utils/set.c
@@ -357,8 +357,12 @@ int main(int argc, char **argv)
 	if (!cpus)
 		cpus = &single_cpu;
 
-	if (related)
-		cpus = cpufreq_get_related_cpus(cpus->cpu);
+	if (related) {
+		struct cpufreq_affected_cpus *tmp;
+		tmp = cpufreq_get_related_cpus(cpus->cpu);
+		if (tmp)
+			cpus = tmp;
+	}
 
 	/* loop over CPUs */
 	while (1) {
-- 
1.8.2.1

